### DISCUSSION TITLE
x-data解析Bug，不支援中括号

### DISCUSSION THREAD

**Discussion by beita-wang** [alibaba/formily#3906] [Q&A 2.x]
往data里传递后端返回的数据，数据的key带中括号，数据会解析错误。
![L15AU4YFBH5NM0V94PE{%5D](https://github.com/alibaba/formily/assets/58249740/8a228f93-e7ef-42ff-be01-a98ff39a4794)
![image](https://github.com/alibaba/formily/assets/58249740/1aa330b3-611a-4248-b3c8-6c6ab711d73a)
小括号就可以~

**Comment by beita-wang** [comment]
后端数据没法更改，只能从前端下手，希望能提供解决办法或者思路~

**Comment by yanhao98** [comment]
存成字符串。
<img width="501" alt="image" src="https://github.com/alibaba/formily/assets/37316281/e4e7ba4f-5c98-4d1c-a809-fc59c4827c18">

**Reply by beita-wang** [reply]
还是不懂, 你试试存到formily的data中，而且formily  schema只能存对象。 我的意思是formily的json schema解析不支援[]

**Reply by yanhao98** [reply]
<img width="448" alt="image" src="https://github.com/alibaba/formily/assets/37316281/259cf06a-d7e2-4204-a7c7-f2728ecba713">
<img width="375" alt="image" src="https://github.com/alibaba/formily/assets/37316281/a19a3e24-4155-438b-a433-93ec46e4410f">


<img width="1538" alt="image" src="https://github.com/alibaba/formily/assets/37316281/7d2dc74d-5d39-4920-9408-e0f0434c6c2a">

https://codesandbox.io/p/sandbox/mystifying-sky-dwwpqw?file=%2Fsrc%2FApp.vue%3A61%2C49

**Reply by yanhao98** [reply]
这样你存的data就是字符串。取出来的时候再JSON.parse()

**Reply by beita-wang** [reply]
你可真是个小机灵鬼。 厉害哟~
